### ISSUE TITLE
Gamepads not always working on "spot" user (amd not working in firefox)

### ISSUE THREAD

**Issue by Dani3I** [vanilla-dpup/woof-CE#2]
I used vanilladpup-11.0.180-labwc-uefi.img.gz , updated via apt and installed heroic games launcher via flatpak (and also nvidia drivers via apt).

When I tried to open Heroic Games Launcher it showed me message in terminal that it won't launch unless I add "--no-sandbox" or run app not as root. So instead I opened Heroic Games Launcher via "run-as-spot". And here comes the wired part: I was able to use gamepad (Xbox) in Heroic Games Launcher (to move around the menu), but when I launched some game, It didn't recognized my gamepad. BUT when I launched Heroic Games Launcher under root account (with "--no-sandbox") gamepad worked fine in Heroic Games Launcher and also in games.

I tried testing gamepad in firefox on https://hardwaretester.com/gamepad and it didn't worked either. Firefox by default is run as spot (I determine that by download folder) but I edited firefox desktop file under /usr/share/applications to not contain "%u" under "Exec=", and I ran that desktop file and it seems that it is running as root after that (again, I guessed that is the case by download folder) and gamepad is not working either.
Chromium on the other hand (installed via apt) is recognizing gamepad under both: root and spot.

I also run jstest-gtk (installed from apt) and it recognizes gamepad under root (normally typing command) and also under spot ("run-as-spot").

BUT I downloaded also supertux-kart (unpacked tar.xz from official website, because there is no apt version for debian trixie) and this game recognizes gamepad only under root (when run via run-as-spot gamepad is not recognized).

**Comment by dimkr** [comment]
Does this still reproduce?

**Comment by Dani3I** [comment]
> Does this still reproduce?

I am probably too inexperienced now to test it properly but it definately doesn't work out-of-the-box. It seems that now (vanilladpup-11.0.206-labwc-uefi.img.gz) vanilladpup have spot user as default user and root is limited. I tried xbox gamepad in firefox and it is not recognized. I tried supertux-kart and also is not recognized.

I tried installing heroic games launcher but I got stuck when trying to launch it:
```
spot@puppypc:/mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64$ flatpak run com.heroicgameslauncher.hgl
error: While opening repository /home/spot/.local/share/flatpak/repo: opening repo: opendir(/home/spot/.local/share/flatpak/repo): Permission denied
```

I also tried to figure out launching as root, found something called pkexec (but I don't know what it is) but I had no luck either:
```
pkexec flatpak run com.heroicgameslauncher.hgl
Run flatpak run com.heroicgameslauncher.hgl as root?
Password: woofwoof
[13 zypak-helper] Failed to connect to session bus: [org.freedesktop.DBus.Error.NoReply] Did not receive a reply. Possible causes include: the remote application did not send a reply, the message bus security policy blocked the reply, the reply timeout expired, or the network connection was broken.
[13 zypak-helper] src/helper/main.cc:42(DetermineZygoteStrategy): Assertion failed: bus
/app/bin/heroic-run: line 6:    13 Aborted                 zypak-wrapper /app/bin/heroic/heroic "$@"
```

also running firefox as root gives:
```
spot@puppypc:/mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64$ pkexec firefox
Run firefox as root?
Password: woofwoof
Running Firefox as root in a regular user's session is not supported.  ($HOME is /home/spot which is owned by spot.)
```

and supertuxkart won't even give any message and do not launches:
```
spot@puppypc:/mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64$ pkexec ./run_game.sh 
Run ./run_game.sh as root?
Password: woofwoof
spot@puppypc:/mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64$ 
```
via bash it gives:
```
spot@puppypc:/mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64$ pkexec bash run_game.sh 
Run bash run_game.sh as root?
Password: woofwoof
bash: run_game.sh: No such file or directory
```
and giving full path it launches as spot:
<details>
output:

```
spot@puppypc:/mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64$ pkexec bash /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/run_game.sh 
Run bash /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/run_game.sh as root?
Password: woofwoof
[verbose  ] main: Error messages and other text output will be logged to /home/spot/.config/supertuxkart/config-0.10/stdout.log.
[info   ] [FileManager]: Data files will be fetched from: '/mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/'
[info   ] [FileManager]: User directory is '/home/spot/.config/supertuxkart/config-0.10/'.
[info   ] [FileManager]: Addons files will be stored in '/home/spot/.local/share/supertuxkart/addons/'.
[info   ] [FileManager]: Screenshots will be stored in '/home/spot/.local/share/supertuxkart/screenshots/'.
[info   ] [FileManager]: User-defined grand prix will be stored in '/home/spot/.local/share/supertuxkart/grandprix/'.
[info   ] [FileManager]: Asset 0 will be loaded from '/mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/challenges/'.
[info   ] [FileManager]: Asset 1 will be loaded from '/mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/gfx/'.
[info   ] [FileManager]: Asset 2 will be loaded from '/mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/grandprix/'.
[info   ] [FileManager]: Asset 3 will be loaded from '/mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/gui/icons/'.
[info   ] [FileManager]: Asset 4 will be loaded from '/mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/gui/screens/'.
[info   ] [FileManager]: Asset 5 will be loaded from '/mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/gui/dialogs/'.
[info   ] [FileManager]: Asset 6 will be loaded from '/mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/replay/'.
[info   ] [FileManager]: Asset 7 will be loaded from '/mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/'.
[info   ] [FileManager]: Asset 8 will be loaded from '/mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/skins/'.
[info   ] [FileManager]: Asset 9 will be loaded from '/mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/ttf/'.
[info   ] [FileManager]: Asset 10 will be loaded from '/mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/po/'.
[info   ] [FileManager]: Asset 11 will be loaded from '/mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/library/'.
[info   ] [FileManager]: Asset 12 will be loaded from '/mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/models/'.
[info   ] [FileManager]: Asset 13 will be loaded from '/mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/music/'.
[info   ] [FileManager]: Asset 14 will be loaded from '/mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/sfx/'.
[info   ] [FileManager]: Asset 15 will be loaded from '/mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/textures/'.
[info   ] [FileManager]: Asset 16 will be loaded from '/mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/tracks/'.
..:: Antarctica Rendering Engine 2.0 ::..
Authorization required, but no authorization protocol specified

[info   ] [IrrDriver Logger]: ..:: Antarctica Rendering Engine 2.0 ::..
Authorization required, but no authorization protocol specified

[info   ] [IrrDriver Logger]: SDL Version 2.24.1
[info   ] [IrrDriver Logger]: Using renderer: OpenGL 4.6
[info   ] [IrrDriver Logger]: Mesa Intel(R) HD Graphics 630 (KBL GT2): Intel
[info   ] [IrrDriver Logger]: OpenGL driver version is 1.2 or better.
[info   ] [IrrDriver Logger]: GLSL version: 4.6
[info   ] IrrDriver: OpenGL version: 4.6
[info   ] IrrDriver: OpenGL vendor: Intel
[info   ] IrrDriver: OpenGL renderer: Mesa Intel(R) HD Graphics 630 (KBL GT2)
[info   ] IrrDriver: OpenGL version string: 4.6 (Core Profile) Mesa 24.2.8-1
[info   ] GLDriver: ARB Buffer Storage Present
[info   ] GLDriver: ARB Compute Shader Present
[info   ] GLDriver: ARB Arrays of Arrays Present
[info   ] GLDriver: ARB Texture Storage Present
[info   ] GLDriver: ARB Texture View Present
[info   ] GLDriver: ARB Image Load Store Present
[info   ] GLDriver: ARB Shader Atomic Counters Present
[info   ] GLDriver: ARB Shader Storage Buffer Object Present
[info   ] GLDriver: EXT Texture Compression S3TC Present
[info   ] GLDriver: ARB Uniform Buffer Object Present
[info   ] GLDriver: ARB Explicit Attrib Location Present
[info   ] GLDriver: EXT Texture Filter Anisotropic Present
[info   ] GLDriver: Geometry Shaders Present
[info   ] GLDriver: ARB Texture Buffer Object Present
[info   ] GLDriver: ARB Texture Swizzle Present
[info   ] GLDriver: ARB Pixel Buffer Object Present
[info   ] GLDriver: ARB Sampler Objects Present
[info   ] GLDriver: ARB Vertex Type 2_10_10_10_rev Present
[info   ] GLDriver: ARB Instanced Arrays Present
[info   ] SharedGPUObjects: Hardware Skinning enabled, method: TBO, max bones: 1024
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/screenquad.vert
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/motion_blur.frag
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/lightning.frag
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/primitive2dlist.vert
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/transparent.frag
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/texturedquad.vert
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/uniformcolortexturedquad.frag
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/texturedquad.frag
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/coloredquad.vert
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/coloredquad.frag
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/colortexturedquad.vert
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/colortexturedquad.frag
[info   ] irr_driver: GLSL supported.
[info   ] GUI: Loading skin data from file: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/skins/peach/stkskin.xml
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/sp_pass.vert
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/sp_solid.frag
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/sp_shadow.vert
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/white.frag
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/sp_skinning.vert
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/sp_skinning_shadow.vert
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/sp_normal_map.frag
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/sp_alpha_test.frag
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/sp_shadow_alpha_test.frag
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/sp_decal.frag
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/sp_grass_pass.vert
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/sp_grass.frag
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/sp_grass_shadow.vert
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/sp_unlit.frag
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/sp_transparent.frag
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/sp_ghost.frag
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/sp_dynamic_night_bloom.frag
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/sp_tilling_mitigation.frag
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/sp_vertical_mapping.frag
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/sp_road_blending.frag
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/sp_displace.frag
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/colorize.frag
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/sp_normal_visualizer.vert
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/sp_normal_visualizer.geom
[info   ] ShaderFilesManager: Compiling shader: /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/shaders/sp_normal_visualizer.frag
XDG_RUNTIME_DIR (/run/runtime) is not owned by us (uid 0), but by uid 1000! (This could e.g. happen if you try to connect to a non-root PulseAudio as a root user, over the native protocol. Don't do that.)
[ALSOFT] (WW) Failed to initialize backend "pulse"
[info   ] GrandPrixManager: Loading Grand Prix files from /mnt/sda8/SuperTuxKart-1.4-linux-x86_64/SuperTuxKart-1.4-linux-x86_64/data/grandprix/
[info   ] GrandPrixManager: Loading Grand Prix files from /home/spot/.local/share/supertuxkart/grandprix/
[warn   ] SPMeshBuffer: kartDirt shader is missing, fallback to solid
[warn   ] SPMeshBuffer: kartDirt shader is missing, fallback to solid
[warn   ] SPMeshBuffer: kartDirt shader is missing, fallback to solid
[warn   ] SPMeshBuffer: kartDirt shader is missing, fallback to solid
[warn   ] SPMeshBuffer: kartDirt shader is missing, fallback to solid
[verbose  ] Thread: Start waiting 59575.662000
[verbose  ] Thread: Waited 0.010000 seconds for thread to become deleteable.
[verbose  ] Thread: Stop waiting 59575.672000
[info   ] Singleton: Destroyed singleton.
[info   ] Singleton: Destroyed singleton.
[info   ] Singleton: Destroyed singleton.
[info   ] Singleton: Destroyed singleton.
[info   ] Singleton: Destroyed singleton.
[info   ] Singleton: Destroyed singleton.
[info   ] Singleton: Destroyed singleton.
[info   ] Singleton: Destroyed singleton.
[info   ] Singleton: Destroyed singleton.
[info   ] Singleton: Destroyed singleton.
warning: queue 0x5637e4828e20 destroyed while proxies still attached:
  wl_callback#31 still attached
  wl_surface#30 still attached
```
</details>

"sudo" seems to be doing nothing and launches things just like spot

**Comment by dimkr** [comment]
First of all, please stop trying to run things as root. Try to `chown -R spot:spot /home/spot`, then please attach any errors you see when you run the game **as spot**.

**Comment by Dani3I** [comment]
Ok I did a little testing and it seems that I know how to solve problem (I hope because I did few other things in meantime but I feel like these shouldn't be relevant).
After running `pkexec chown -R spot:spot /home/spot` and also restarting system, heroic games launcher finally launched. So either restart or your command worked.

Then game-pad worked in heroic menu, and I also installed jstest-gtk and gamepad also worked here. (But game-pad still didn't worked in firefox and supertuxkart)

I did a little more tests and looked online and ultimately I edited "/etc/group" file (`pkexec geany /etc/group`): I added "spot" to "input" group (changed line `input:x:997:` to `input:x:997:spot`). And after that I restarted system.

Game-pad started working in supertuxkart, but firefox (https://hardwaretester.com/gamepad ) still didn't detected game-pad.
So I downloaded standalone package of firefox from official website (.tar.bz2 file - this one is downloaded by default on vanilladpup), and game-pad started working in firefox. So in preinstalled firefox game-pad is not detected, but in firefox downloaded from official website it works.

I also wanted to test games from heroic games launcher, the same way I mentioned I did in my initial message but unfortunately I have problems with logging in into account in heroic for some reason (in browser I have no troubles).
